core/src/apps/cardano/layout.py

```diff
@@ -331,6 +357,57 @@ async def show_warning_tx_output_contains_tokens(ctx: wire.Context) -> None:
     )
 
 
+async def show_warning_tx_output_contains_datum_hash(
+    ctx: wire.Context, datum_hash: bytes
+) -> None:
+    await confirm_properties(
+        ctx,
+        "confirm_datum_hash",
+        title="Confirm transaction",
+        props=[
+            (
+                "The following\ntransaction output\ncontains datum hash:",
+                datum_hash,
+            ),
+            ("\nContinue?", None),
+        ],
+        br_code=ButtonRequestType.Other,
+    )
+
+
+async def show_warning_tx_contains_mint(ctx: wire.Context) -> None:
+    await confirm_metadata(
+        ctx,
+        "confirm_tokens",
+        title="Confirm transaction",
+        content="The transaction contains\nminting or burning of\ntokens.",
+        larger_vspace=True,
+        br_code=ButtonRequestType.Other,
+    )
+
+
+async def show_warning_no_script_data_hash(ctx: wire.Context) -> None:
+    await confirm_metadata(
+        ctx,
+        "confirm_no_script_data_hash",
+        title="Confirm transaction",
+        content="The transaction contains\nno script data hash.\n"
+        "Plutus script will not be\nable to run.",
```

Comment:
Why is this on a new line? Let's add it to the previous line string and let's also remove the `\n`s.

Reply:
Ok, fixed (I originally put it on two lines to not exceed line length limit).